### DISCUSSION TITLE
Option to override arbitrary tags

### DISCUSSION THREAD

**Discussion by gedewon** [markedjs/marked#3561] [Q&A] [answered]
Hi team , 
i am wondering if there is a way to override any html tag or any arbitrary tags like [marked-to-jsx](https://github.com/quantizor/markdown-to-jsx?tab=readme-ov-file#optionsoverrides---rendering-arbitrary-react-components) does ? and then override or define a rendere .

example markdown:
   ```markdown
  <ButtonLink>decline all non-essential cookies</ButtonLink> 
  <p>if you do not agree with the data processing described.<p>
```

custom renderer: 
```jsx
const renderer: RendererObject = {
    ButtonLink({ tokens, depth }) {
        // custom implementation
        return <ButtonLink onClick={handleClick}></ButtonLink>;
    },
};

```

 i see we only have pre defined set of renderer method and can't extend it to use custom or non html tags ? https://github.com/markedjs/marked/blob/10020d018d4448b1d2d4612baf650c4e3a5949bf/src/Renderer.ts#L14

Thanks in advance!!

**Comment by UziTech** [comment, marked as answer]
Yes you can create a [custom extension](https://marked.js.org/using_pro#extensions) that can tokenize any text you specify and render it however you want. 

If you want to be able to return something other than an html string you can use the `provideParser` hook.

**Reply by gedewon** [reply]
Amazing thanks !!
